connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,112 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.*;
+import java.lang.reflect.InvocationTargetException;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConnectionTest {
```

Comment:
&#x1F34E; **Issue**
Could this method be split into two more granular methods (and be used granularry in your @Test cases) ?
May be you don't need to elaborately set up "innerInputStream" when running "shouldSendMessageWhenAskingForIt" @Test (and similar cases like this one hold true for other tests) ?